feature_engine/encoding/woe.py

```diff
@@ -151,40 +151,36 @@ def fit(self, X: pd.DataFrame, y: pd.Series):
                 "used has more than 2 unique values."
             )
 
-        self._fit(X)
-        self._get_feature_names_in(X)
-
-        temp = pd.concat([X, y], axis=1)
-        temp.columns = list(X.columns) + ["target"]
-
         # if target does not have values 0 and 1, we need to remap, to be able to
         # compute the averages.
         if any(x for x in y.unique() if x not in [0, 1]):
-            temp["target"] = np.where(temp["target"] == y.unique()[0], 0, 1)
+            y = y.map({y.unique()[0]: 0, y.unique[1]:1})
+
+        self._fit(X)
+        self._get_feature_names_in(X)
 
         self.encoder_dict_ = {}
 
-        total_pos = temp["target"].sum()
-        total_neg = len(temp) - total_pos
-        temp["non_target"] = np.where(temp["target"] == 1, 0, 1)
+        total_pos = y.sum()
+        total_neg = len(y) - total_pos
+        inverse_y = y.map({1: 0, 0:1}).copy()
```

Comment:
arithmetical operations would be faster than mapping, we can do it like this:
`total_pos = y.sum()`
`inverse_y = (y - 1).abs().copy()`
`total_neg = inverse_y.sum()`
Should be faster

Reply:
Thank you @GLevV 

I also replaced the previous mapping by scaling :)